### PR TITLE
feat: improve key lookup

### PR DESCRIPTION
**Pull request by Stebalien**
* Support peer IDs encoded as CIDs.
* Support looking up the identity key by peer ID.

### CODE REVIEW

File: core/coreapi/name.go [hunk 1]
```diff
@@ -170,7 +194,7 @@ func keylookup(self ci.PrivKey, kstore keystore.Keystore, k string) (ci.PrivKey,
 			return nil, err
```
Comment (hsanjuan) [inline core/coreapi/name.go]: Why is it not doing IDFromPrivateKey(privKey) directly?

File: core/coreapi/name.go [hunk 2]
```diff
@@ -157,6 +163,24 @@ func keylookup(self ci.PrivKey, kstore keystore.Keystore, k string) (ci.PrivKey,
...
+	//////////////////
+	targetPid, err := peer.Decode(k)
+	if err != nil {
+		return nil, keystore.ErrNoSuchKey
+	}
+
+	// First, check self.
+	pid, err := peer.IDFromPrivateKey(self)
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine peer ID for private key")
```
Comment (hsanjuan) [inline core/coreapi/name.go]: Not including the original error message?

File: core/coreapi/name.go (hunk 1 above)
Comment (hsanjuan) [reply core/coreapi/name.go]: Just a couple of lines above, could not add comment directly there.

Comment (Stebalien) [reply core/coreapi/name.go]: No idea. It may not have existed? I'll change that.

**Pull request merged**